packages/components/src/utils/getNamedSlot.ts

```diff
@@ -0,0 +1,5 @@
+import { getHTMLElement } from './dom';
+import type { HTMLElementOrShadowRoot } from './dom';
+
+export const getNamedSlot = (el: HTMLElementOrShadowRoot, slotName: string): HTMLSlotElement =>
```

Comment:
I think this return type is wrong. You are selecting something by the `slot` attribute which can never be a `slot` element.